rust/kernel/str.rs

```diff
@@ -45,6 +45,7 @@ pub enum CStrConvertError {
 }
 
 impl From<CStrConvertError> for Error {
+    #[cold]
```

Comment:
This doesn't make sense because the function is already marked as `inline`.

In fact I think a lot of these small functions should be marked as `inline` instead.

Reply:
Even if it is unused for LLVM nowadays, could it be useful in the future, or for documentation purposes, or maybe for other backends like GCC?

Reply:
GCC indeed could use this information (I tested `__attribute__((cold)) inline` in C and it seems to both hint the unlikeliness and get inlined).

BTW: Is error actually unlikely? For IO stuff I suppose EAGAIN is actually quite frequent.